components/tikv_util/src/sys/cgroup.rs

```diff
@@ -0,0 +1,463 @@
+        if field == OPTIONAL_FIELDS_SEP {
+            found_sep = true;
+            break;
+        }
+        sep_pos += 1;
+    }
+    if !found_sep {
+        return Err(
+            "mount point format invalid, doesn't found optional field separator".to_string(),
+        );
+    }
+
+    let fs_start = sep_pos + 1;
+    let mount_id = fields[MountInfoFieldPart1::MountID as usize]
+        .parse::<i32>()
+        .unwrap();
+    let parent_id = fields[MountInfoFieldPart1::ParentID as usize]
+        .parse::<i32>()
+        .unwrap();
+    let mount_point = MountPoint {
+        mount_id,
+        parent_id,
+        device_id: fields[MountInfoFieldPart1::DeviceID as usize].clone(),
+        root: fields[MountInfoFieldPart1::Root as usize].clone(),
+        mount_point: fields[MountInfoFieldPart1::MountPoint as usize].clone(),
+        options: fields[MountInfoFieldPart1::Options as usize]
+            .split(OPTIONS_SEP)
+            .map(|s| s.to_string())
+            .collect(),
+        optional_fields: fields[MountInfoFieldPart1::OptionalFields as usize..fs_start - 1]
+            .to_vec(),
+        fs_type: fields[fs_start + MountInfoFieldPart2::FSType as usize].clone(),
+        mount_source: fields[fs_start + MountInfoFieldPart2::MountSource as usize].clone(),
+        super_options: fields[fs_start + MountInfoFieldPart2::SuperOptions as usize]
+            .split(OPTIONS_SEP)
+            .map(|s| s.to_string())
+            .collect(),
+    };
+
+    Ok(mount_point)
+}
+
+enum SubsysFields {
+    ID = 0,
+    Subsystems,
+    Name,
+
+    Count,
+}
+
+#[derive(Clone, Debug, PartialEq)]
+pub struct CGroupSubsys {
+    pub id: i32,
+    pub sub_systems: Vec<String>,
+    pub name: String,
+}
+
+pub fn parse_subsys_from_line(line: &str) -> Result<CGroupSubsys, Error> {
+    let fields: Vec<String> = line.split(CGROUP_SEP).map(|s| s.to_string()).collect();
+    if fields.len() != SubsysFields::Count as usize {
+        return Err(box_err!("subsystem format invalid"));
+    }
+
+    let id = fields[SubsysFields::ID as usize].parse::<i32>().unwrap();
+    Ok(CGroupSubsys {
+        id,
+        sub_systems: fields[SubsysFields::Subsystems as usize]
+            .split(SUBSYS_SEP)
+            .map(|s| s.to_string())
+            .collect(),
+        name: fields[SubsysFields::Name as usize].clone(),
+    })
+}
+
+pub fn file_scanner<F>(path: &str, mut f: F) -> std::io::Result<()>
+where
+    F: FnMut(&str),
+{
+    let file = File::open(path)?;
+    let reader = BufReader::new(file);
+    for line in reader.lines() {
+        if let Ok(l) = line {
+            f(&l);
+        }
+    }
+    Ok(())
+}
+
+#[derive(Debug, PartialEq)]
+pub struct CGroup {
+    // subsystem path
+    path: String,
+}
+
+impl CGroup {
+    pub fn new(path: String) -> Self {
+        Self { path }
+    }
+
+    pub fn read_num(&self, param: &str) -> Result<i64, Box<dyn error::Error>> {
+        Ok(fs::read_to_string(Path::new(&self.path).join(param))?
```

Comment:
Seems this logic is different from https://github.com/uber-go/automaxprocs/blob/master/internal/cgroups/cgroup.go#L73.  I think we just need read the first line and prevent the multiple lines file volatile our expectations.